accumulator_updater/programs/mock-cpi-caller/src/instructions/add_price.rs

```diff
@@ -0,0 +1,129 @@
+use {
+    crate::{
+        instructions::sighash,
+        message::{
+            get_schemas,
+            price::{
+                CompactPriceMessage,
+                FullPriceMessage,
+            },
+            AccumulatorSerializer,
+        },
+        state::{
+            PriceAccount,
+            PythAccountType,
+        },
+    },
+    accumulator_updater::program::AccumulatorUpdater as AccumulatorUpdaterProgram,
+    anchor_lang::{
+        prelude::*,
+        solana_program::sysvar,
+    },
+};
+
+pub fn add_price<'info>(
+    ctx: Context<'_, '_, '_, 'info, AddPrice<'info>>,
+    params: AddPriceParams,
+) -> Result<()> {
+    let mut account_data: Vec<Vec<u8>> = vec![];
+    let schemas = get_schemas(PythAccountType::Price);
+
+    {
+        let pyth_price_acct = &mut ctx.accounts.pyth_price_account.load_init()?;
+
+        pyth_price_acct.init(params)?;
+
+        let price_full_data = FullPriceMessage::from(&**pyth_price_acct).accumulator_serialize()?;
+
+        account_data.push(price_full_data);
+
+
+        let price_compact_data =
+            CompactPriceMessage::from(&**pyth_price_acct).accumulator_serialize()?;
+        account_data.push(price_compact_data);
+    }
+
+
+    let account_schemas = schemas.into_iter().map(|s| s.to_u8()).collect::<Vec<u8>>();
+
+    // 44444 compute units
+    // AddPrice::invoke_cpi_anchor(ctx, account_data, PythAccountType::Price, account_schemas)
+    // 44045 compute units
+    AddPrice::emit_accumulator_inputs(ctx, account_data, PythAccountType::Price, account_schemas)
```

Comment:
oh, I think this question may come down to a misunderstanding of how the oracle program works. In the oracle program, when you add a price account, it doesn't actually contain any price information. it simply allocates the space for putting prices in later. So there wouldn't even be any data to emit. (whereas this version of add price actually puts some data in the price account).

anyway, since this is just a test program, so I would scope it down to the minimum necessary code to demonstrate how to invoke the accumulators & test all the needed functionality. pretty sure you don't need both add price and update price to do that.

Reply:
oh got it thanks for clarifying that. i'll update this and the relevant tests/code and scope it down

Reply:
I left the `add_price` & cpi call in for now but left a comment describing the actual behavior in the oracle program.